### ISSUE TITLE
Ordering metadata fields with multiple values needs discussion

### ISSUE THREAD

**Issue by jlhardes** [samvera/hyrax#4781]
### Descriptive summary

Metadata ordering when there are multiple values for one field (Creator, for example) is important in certain contexts, such as scholarly communication. It is not important in all contexts where Hyrax is used (seems like something close to a digital collections vs institutional repository use case difference but it is probably more nuanced than that).

Currently there is no single solution from Hyrax for handling order across multiple values for the same metadata field. The impact of this is that with no order specified and if metadata is stored as RDF (which is the case for Hyrax-based applications using Fedora as the backend), the order of these multiple values can be different each time an item is viewed. 

Local implementations have addressed metadata field order in various ways for various purposes. Some of these implementations are identified in the report from the [Hyrax Metadata Ordering Working Group](https://wiki.duraspace.org/download/attachments/90977535/Hyrax_Metadata_Ordering_Report.pdf) as well as more recently from [Rutgers via Notch8](https://github.com/dibbs-vdc/ccql/blob/master/app/models/vdc/resource.rb#L132) and [University of Michigan](https://fritzfreiheit.com/wiki/Adding_Ordered_Metadata_Fields_to_Samvera_Hyrax).

Google Scholar has reported problems from Hyrax-based implementations with author order, in particular, changing in the "citation_[xxx]" <meta> tags they use to identify articles for indexing. Author order in this context is crucial to be accurate and consistent. More info on Google Scholar meta-tags: https://scholar.google.com/intl/en/scholar/inclusion.html#indexing An example in Nurax-dev with multiple Creators: https://nurax-dev.curationexperts.com/concern/generic_works/r494vk36q?locale=en

Google Scholar meta-tags have been generated for all Resource Types in Hyrax. The linked pull request (#4774) makes changes to suppress Google Scholar meta-tags on non-scholarly content.

@samvera/hyrax-metadataists, questions to consider:
* What is the desired behavior for Hyrax regarding order for metadata fields with multiple values? 
  * When should metadata fields with multiple values be ordered or when should they be able to have an order specified within Hyrax?
  * Are there particular Resource Types where Creator order should be definable?

### Related work

Also related to issue #110

**Comment by noraegloff** [comment]
SMIG discussion of this issue is still ongoing, but three points came out of our meeting today: 

- **Creator** and **Contributor** are two fields that in particular would benefit from ordered multiple values. Other fields that this could potentially apply to/be helpful for would be **Subject**, **Keyword**, and **Title**.  

- There are many different contexts where multiple value metadata ordering could be important, for example scholarly communication and archives/special collections use cases.  Since there isn't a predictable ordering for each context that could be anticipated and automated, a **manual** way of indicating order within the interface would be best. 

- The utility of multiple value metadata ordering is not necessarily limited to specific Resource Types.

**Comment by wickr** [comment]
Some impact relation to https://github.com/samvera/hyrax/issues/395

**Comment by wickr** [comment]
Strongly related to https://github.com/samvera/hyrax/issues/1568

**Comment by agoslen** [comment]
SMIG discussed this again on [April 27](https://samvera.atlassian.net/wiki/spaces/samvera/pages/1318060033/Metadata+Call+2021-04-27). 

- There is continued agreement that the value of metadata ordering for multi-valued fields is not limited to particular Resource Types.
- SMIG recommends that **Creator**, **Contributor**, and **Title** are the fields where ordering is most important

**Comment by shlake** [comment]
In the original description "A recent pull request (#4774) makes changes to suppress Google Scholar meta-tags on non-scholarly content"

How is "non-scholarly" content distinguished between "scholarly" content?

**Comment by jlhardes** [comment]
I think the idea is to use Resource Types to define scholarly vs non-scholarly but I don't know if any Resource Types have actually been declared one way or another, at least in Hyrax out of the box.

**Comment by rjkati** [comment]
In Hyrax 3.0.2, fields which allow multiple values are not ordered: https://nurax-dev.curationexperts.com/concern/generic_works/w9505077h?locale=en

**Comment by sandbergja** [comment]
Our repository isn't very scholarly; it's mainly open educational resources and institutional history.  Despite this, we still get requests for the creator and contributor fields to be in a particular meaningful order.  For example, we just added an [anthology of works by our student poet laureates](https://libarchive.linnbenton.edu/concern/documents/3484zh25j).  We got a request that the contributors (the student poet laureates) be listed chronologically, from the first student to hold that position to the most recent.  This is also the order they are listed in the work itself. :-)

We have also gotten requests for the following fields:

- keyword and subject.  Sometimes the order is to show which topics are more important; other times it is the order that they are discussed in the work
-  language: for multilingual works, our users want to put either the original or the more prominent language first

**Comment by midnitelibrary** [comment]
Hi there, just wondering if anything has happened with this? We have a faculty member who wants the creators listed so that they match the citation order.

**Comment by rjkati** [comment]
@midnitelibrary, there hasn't been any movement on this, but we hope to address it in the future. I do appreciate your comment, it is always good to know that folks are still interested in the enhancement.

**Comment by acklose** [comment]
@rjkati Yes, this is still of interest to members of SMIG. We were impressed to see the customization UNC has for ordering authors' names.

**Comment by acklose** [comment]
SMIG is still very interested in ordering multiple values in a metadata field, e.g. Creator. However, ordering might be achieved with [order already gem](https://github.com/samvera-labs/order_already). In Valkyrie with Postgres, there is reportedly ordering of values but it just happens and is not tested. Reordering of values is copy and paste. FYI - UNC has their own way to address this (see [pull request in their GitHub repo](https://github.com/UNC-Libraries/hy-c/pull/489/files)).
